zgrab2/processing.go

```diff
@@ -0,0 +1,126 @@
+package zgrab2
+
+import (
+	"bufio"
+	"encoding/json"
+	"io"
+	"net"
+	"sync"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type Grab struct {
+	IP     *string                   `json:"ip,omitempty"`
```

Comment:
Why is this a pointer? It's exceedingly easy to have concurrency bugs with pointers.